tests/unittests/cmd/test_main.py

```diff
@@ -1,75 +1,74 @@
 # This file is part of cloud-init. See LICENSE file for license information.
 
 import copy
+import getpass
 import os
 from collections import namedtuple
-from io import StringIO
 from unittest import mock
 
 import pytest
 
 from cloudinit import safeyaml
 from cloudinit.cmd import main
 from cloudinit.util import ensure_dir, load_text_file, write_file
-from tests.unittests.helpers import FilesystemMockingTestCase, wrap_and_call
 
 MyArgs = namedtuple(
     "MyArgs", "debug files force local reporter subcommand skip_log_setup"
 )
 
 
-class TestMain(FilesystemMockingTestCase):
-    with_logs = True
-    allowed_subp = False
-
-    def setUp(self):
-        super(TestMain, self).setUp()
-        self.new_root = self.tmp_dir()
-        self.cloud_dir = self.tmp_path("var/lib/cloud/", dir=self.new_root)
-        os.makedirs(self.cloud_dir)
-        self.replicateTestRoot("simple_ubuntu", self.new_root)
-        self.cfg = {
+class TestMain:
+    @pytest.fixture(autouse=True)
+    def common_mocks(self, mocker):
+        mocker.patch("cloudinit.cmd.main.close_stdin")
+        mocker.patch(
+            "cloudinit.cmd.main.netinfo.debug_info",
+            return_value="my net debug info",
+        )
+        mocker.patch(
+            "cloudinit.cmd.main.util.fixup_output",
+            return_value=("outfmt", "errfmt"),
+        )
+        mocker.patch("cloudinit.cmd.main.util.get_cmdline", return_value="")
+        mocker.patch("cloudinit.cmd.main.util.uptime", return_value="12345")
+        os.environ["_CLOUD_INIT_SAVE_STDOUT"] = "true"
+        yield
+        os.environ.pop("_CLOUD_INIT_SAVE_STDOUT")
+
+    @pytest.fixture
+    def cloud_cfg(self, mocker, tmpdir, fake_filesystem):
```

Comment:
I had wondered about using our paths fixture from conftest.py here. But, because of the need to ensure_dir throughout it didn't make sense.